### PR TITLE
Refactoring and API-Support

### PR DESCRIPTION
**Pull request by JW-CH**

### CODE REVIEW

**Comment by JW-CH** [comment]
@3rob3 let me know if anything more is needed. I think this should be enough to have the base set for API-Support.

**Comment by 3rob3** [comment]
Been gone all weekend (holiday weekend here). I will take a look early this week, sorry for delay.

**Comment by JW-CH** [comment]
It has been less than a day. Take your time. No worries!

File: ImmichFrame.WebApi/Program.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+using ImmichFrame.Core.Exceptions;
+using ImmichFrame.Core.Interfaces;
+using ImmichFrame.WebApi.Models;
+using System.Text.Json;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+var settingsPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Settings.json");
```
Comment (3rob3) [inline ImmichFrame.WebApi/Program.cs:10]: What is the plan for the settings file? Hard coded into the "server" during Docker creation?

Comment (JW-CH) [reply ImmichFrame.WebApi/Program.cs:10]: Either this or via env variables

**Pull request merged**